Assets/Scripts/Utilities/AudioZone.cs

```diff
@@ -0,0 +1,31 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AudioZone : MonoBehaviour 
+{
+	[SerializeField] AudioSource sourceData = null;
+	AudioSource playingClip = null;
+
+	void OnTriggerEnter( Collider col )
+	{
+		if ( col.GetComponent<PlayerActor>() )
+		{
+			playingClip = SoundManager.instance.Play3DSoundAndFollow( sourceData, transform );
+		}
+	}
+
+	void OnTriggerExit( Collider col )
+	{
+		if ( col.GetComponent<PlayerActor>() && playingClip.gameObject )
```

Comment:
Why check `playingClip.gameObject` for null and not just `playingClip`? Is there ever a situation where a component would exist but its game object wouldn't?